WordPress/WordPressShareExtension/ShareExtractor.swift

```diff
@@ -128,19 +139,35 @@ private extension TypeBasedExtensionContentExtractor {
         return !context.itemProviders(ofType: acceptedType).isEmpty
     }
 
-    func extract(context: NSExtensionContext, completion: @escaping (ExtractedItem?) -> Void) {
-        guard let provider = context.itemProviders(ofType: acceptedType).first else {
+    func extract(context: NSExtensionContext, completion: @escaping ([ExtractedItem]) -> Void) {
+        let itemProviders = context.itemProviders(ofType: acceptedType)
+        guard itemProviders.count > 0 else {
             DispatchQueue.main.async {
-                completion(nil)
+                completion([ExtractedItem]())
             }
             return
         }
-        provider.loadItem(forTypeIdentifier: acceptedType, options: nil) { (payload, error) in
-            let payload = payload as? Payload
-            let result = payload.flatMap(self.convert(payload:))
 
-            DispatchQueue.main.async {
-                completion(result)
+        // There 1 or more valid item providers here, lets work through them
+        var results = [ExtractedItem]()
+        var itemsToLoad = 0
+        for provider in itemProviders {
+            itemsToLoad += 1
+            // Remember, this is an async call....
+            provider.loadItem(forTypeIdentifier: acceptedType, options: nil) { (payload, error) in
+                let payload = payload as? Payload
+                let result = payload.flatMap(self.convert(payload:))
+                if let result = result {
+                    results.append(result)
+                }
+                itemsToLoad -= 1
+
+                // This only gets called when _all_ of the items are loaded
+                if itemsToLoad == 0 {
```

Comment:
I think `DispatchGroup` is a better fit for this and thread safe. In theory, the first `loadItem` could complete before the following one is started, so `itemsToLoad -=1` would be 0 and it could call the completion block.